### ISSUE TITLE
Just because you thought no one would ask :)

### ISSUE THREAD

**Issue by MHarris021** [alasdairforsythe/pansearch#1]
Can you please add usage instructions? :)

**Comment by cryolite-ai** [comment]
@MHarris021 I came upon the same thing as I was catching up with some LLM research and noticed the TokenMonster work; to save @alasdairforsythe the effort, since the code is in a single Go package, I threw the whole of it into an LLM (Claude.ai), asked it to create a summary of the usage of the package. After a second prompt to get it to revise its output, here is the result.. if it's accurate then perhaps it could be put into the main documentation and the issue closed ;-). This approach will benefit from, ahem, more comments in code.. but clear variable name choice will make things easier for the LLM to produce something more likely to be meaningful.

Hope this helps you both.


--------------------------------------------------------------------------------------

Here is a revised overview and usage instructions for the pansearch package:

The pansearch package provides binary search indexes for byte slice keys in Go. It offers several index implementations optimized for different use cases:

The pansearch package aims to enable efficient storage and retrieval of keys in applications that need to handle large volumes of data. The keys can be of arbitrary byte lengths up to 64 bytes. 

The package provides the following index implementations:

- **Fast** - Optimized for fast lookups while maintaining moderate memory usage. Good for read-heavy use cases.

- **Light** - Minimizes memory usage for large indexes while still allowing efficient lookups and iteration.

- **Counter** - Stores counts for each key. Useful for aggregating and filtering by value.

To use pansearch, import the package:

```go 
import "github.com/AlasdairF/BinSearch/pansearch"
```

Here is an example of using the Fast index:

```go
index := pansearch.Fast{}

index.Add([]byte("key1"))
index.Add([]byte("key2"))

index.Build()

idx1, exists := index.Find([]byte("key1"))
idx2, exists := index.Find([]byte("key2")) 

index.Reset() 
for {
  key, eof := index.Next()
  if eof {
    break
  }
  // Use key
}
```

The Light index usage is similar, but keys are added with `AddUnsorted()` instead of `Add()`, and the index is built with `Build()`.

The Counter index supports aggregating values per key:

```go
index := pansearch.Counter{}

index.Add([]byte("key1"), 1)
index.Add([]byte("key2"), 5) 

index.Build()

index.Update([]byte("key1"), func(count int) int {
  return count + 1
})

index.Build(10) // Filter keys with count >= 10

for {
  key, count, eof := index.Next()
  if eof {
    break
  }
  // Use key and count
}
```

So in summary, pansearch provides optimized binary search indexes for byte slice keys in Go applications.
